default.nix

```diff
@@ -0,0 +1,221 @@
+      (shareTable 26 30 7 5)
+      recovery
+      fusionInner
+      fusionOuter
+      checksumTable1
+      checksumTable2
+      checksumWorksheet
+    ];
+  };
+
+  dependencyContentRecur = content: builtins.concatMap
+    (item: (dependencyContentRecur item.dependencies) ++ [ item.content ])
+    content;
+  dependencyContent = pages: lib.lists.unique (
+    (map (dep: dep.content) standardDependencies) ++
+    (builtins.concatMap (page: dependencyContentRecur page.dependencies) pages)
+  );
+
+  renderBooklet = booklet:
+    let
+      addPage = content: pageData: {
+        content = content.content + lib.optionalString (pageData ? sourceHeader) ''
+          %****************************************************************
+          %*
+          %* ${pageData.sourceHeader}
+          %*
+          %****************************************************************
+        '' + ''
+          %%Page: ${toString content.nextPgIdx} ${toString content.nextPgIdx}
+          ${lib.optionalString (pageData ? isLandscape) "%%PageOrientation: Landscape\n"}%%BeginPageSetup
+          /pgsave save def
+          %%EndPageSetup
+        '' + (
+          if pageData ? drawPageContent
+          then
+            if pageData ? isLandscape
+            then "landscapePage begin ${toString content.nextFooterIdx} drawPageContent\n"
+            else "portraitPage begin ${toString content.nextFooterIdx} drawPageContent\n"
+          else
+            if pageData ? isLandscape
+            then "90 rotate\n"
+            else ""
+        ) + ''
+          ${pageData.content}
+          ${lib.optionalString (pageData ? drawPageContent) "end\n"}pgsave restore
+          showpage
+        '';
+        nextFooterIdx = content.nextFooterIdx + (if pageData ? drawPageContent then 1 else 0);
+        nextPgIdx = content.nextPgIdx + 1;
+      };
+      initialContent = {
+        content = ''
+          %!PS-Adobe-3.0
+          %%Orientation: Portrait
+          %%Pages: ${toString (builtins.length booklet.pages)}
+          %%EndComments
+          %%BeginSetup
+          ${toString (dependencyContent (booklet.pages))}%%EndSetup
+
+          %************************************************************************
+          %************************************************************************
+          %*
+          %* Section Three: Page Rendering
+          %*
+          %************************************************************************
+          %************************************************************************
+
+        '';
+        nextPgIdx = 1;
+        nextFooterIdx = 1;
+      };
+      finalContent = builtins.foldl' addPage initialContent booklet.pages;
+    in
+    pkgs.writeTextFile {
+      name = booklet.name;
+      text = finalContent.content + ''
+        %%EOF
+      '';
+    };
+in
+stdenv.mkDerivation {
+  name = "codex32${shortId}";
+
+  buildInputs = if doPdfGeneration then [ ghostscript ] else [ ];
```

Comment:
I think this ought to be `nativeBuildInputs`.

Reply:
Ok, will change. I really don't understand the distinction.

Reply:
My vague understanding is that `buildInputs` is historic, and means build and runtime dependencies.  While `nativeBuildInputs` is just buildtime dependencies.  I'm not sure what the actual effect of this difference is.